### DISCUSSION TITLE
Setting cursor visibility or color

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [tui-cs/Terminal.Gui#2869] [Q&A] [answered]
In my app the blinking cursor is very hard to see in text fields. After reading the docs, I thought I might try setting the cursor to a box.

```powershell
$txtDays = [Terminal.Gui.TextField]@{
        Width   = 4
        Text    = 30
        Y       = $txtDescription.Y + 2
        X       = $radioGrp.Frame.Width + 2
        TabStop = $True
        DesiredCursorVisibility = "Box"
    }
```
But this doesn't seem to have any effect. Unless the cursor is a box with the same color as the text field. Any tips on defining a contrasting cursor?

**Comment by BDisp** [comment]
Maybe you have to change the cursor setting on the terminal. I experienced that issue a long tie ago and I have to set the cursor format to box on the `Windows Terminal`.

**Reply by jdhitsolutions** [reply, marked as answer]
That's exactly what it was. The form was inheriting the cursor setting from Windows Terminal. Setting `DesiredCursorVisibility` doesn't have any effect in this environment. Although I still wish I could change the color.

**Reply by BDisp** [reply]
We can't change the color. It's terminal specific. With `Terminal.Gui` with `Windows OS` we can change the visibility, box and underline settings. With `xterm` we can change the additional settings:
```
Default
Vertical
Vertical Fix
Box Fix
Underline Fix
```

**Reply by jdhitsolutions** [reply]
I addressed the potential issue in my documentation.
